03-pipefilter.md

```diff
@@ -79,119 +86,86 @@ $ wc *.pdb
 > themselves. It is the shell, not the other programs, that deals with
 > expanding wildcards, and this is another example of orthogonal design.
 
-If we run `wc -l` instead of just `wc`,
-the output shows only the number of lines per file:
-
-~~~ {.bash}
-$ wc -l *.pdb
-~~~
-~~~ {.output}
-  20  cubane.pdb
-  12  ethane.pdb
-   9  methane.pdb
-  30  octane.pdb
-  21  pentane.pdb
-  15  propane.pdb
- 107  total
-~~~
-
-We can also use `-w` to get only the number of words,
-or `-c` to get only the number of characters.
-
-Which of these files is shortest?
-It's an easy question to answer when there are only six files,
-but what if there were 6000?
-Our first step toward a solution is to run the command:
+The first question that we can ask is
+"how many files are inside the `molecules` directory?".
+If we could (1) write the output of `ls *.pdb` into a file
+and (2) count the number of lines of that file
+we will have our answer.
+We are very luck and we can do (1) and (2) with the shell.
+For (1) we will use
 
 ~~~ {.bash}
-$ wc -l *.pdb > lengths.txt
+$ ls *.pdb > molecule-list
```

Comment:
I would suggest using the best-practice of adding extensions also here, so using `molecule-list.txt` as filename